pulp_ansible/app/galaxy/views.py

```diff
@@ -154,18 +155,25 @@ def post(self, request, path):
         )
         serializer.is_valid(raise_exception=True)
 
-        artifact = Artifact.init_and_validate(serializer.validated_data["file"])
-        artifact.save()
+        temp_file = serializer.validated_data["file"]
+        temp_file_path = os.path.join(settings.WORKING_DIRECTORY, f"pulp_upload_{temp_file.name}")
+        with open(temp_file_path, "wb") as temp_import_file:
+            shutil.copyfileobj(temp_file, temp_import_file)
 
-        locks = [str(artifact.pk)]
-        kwargs = {"artifact_pk": artifact.pk}
-        if distro.repository:
-            locks.append(distro.repository)
-            kwargs["repository_pk"] = distro.repository.pk
-
-        async_result = enqueue_with_reservation(import_collection, locks, kwargs=kwargs)
+        async_result = self._dispatch_import_collection_task(distro.repository, temp_file_path)
         return OperationPostponedResponse(async_result, request)
 
+    def _dispatch_import_collection_task(self, repository, temp_file_path, **kwargs):
```

Comment:
Can we dedpulicate this from being defined in multiple places with a mixin object? Let me know if this suggestion needs more details to implement.